crates/ruff_python_formatter/src/expression/parentheses.rs

```diff
@@ -53,6 +53,30 @@ pub enum Parenthesize {
 
     /// Parenthesizes the expression only if it doesn't fit on a line.
     IfBreaks,
+
+    /// This is special case for StmtFor where we remove parentheses.
+    ///
+    /// In most cases, we keep the parentheses, e.g.
+    /// ```python
+    /// (a, b) = x
+    /// a, b = x
+    /// ```
+    /// both remain as-in. The exception are for loops:
+    /// ```python
+    /// for a, b in x:
+    ///     pass
+    /// for (a, b) in x:
+    ///     pass
+    /// ```
+    /// is formatted as
+    /// ```python
+    /// for a, b in x:
+    ///     pass
+    /// for a, b in x:
+    ///     pass
+    /// ```
+    /// . That is, we remove the parentheses unless the group breaks
+    RemoveUnlessBreaks,
```

Comment:
How is this different from `IfBreaks`? `IfBreaks` only adds parentheses if the group breaks but removes them otherwise.

Reply:
Isn't the same true for `IfStmt` where we remove parentheses in the condition (it uses `IfBreaks`) (and all match cases with an expression)? 

I believe `IfBreaks` already gives you what you need, with the exception that the current `NeedsParentheses` should return `Custom` if `default_needs_parentheses` returns `Parentheses::Optional` instead of `Never`